### PR TITLE
Sonarlint update to JsonMergePatchHttpMessageConvertor

### PR DESCRIPTION
**Pull request by CanadianPhilipBarkow**
### Description

Updated `JsonMergePatchHttpMessageConvertor` based on sonarlint warnings.

- Added try-with-resources to close `JsonReader` after use.
- Added try-with-resources to close `JsonWriter` after use.
- Relevant test updates made.

### Related Azure Boards Work Items

[AB#4173](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4173)
[AB#4180](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4180)
[AB#4181](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4181)

### Checklist

- [X] I have tested the changes locally
- [X] I have added/updated tests that prove my fix is effective or that my feature works
- [X] I have checked that my code follows the project's coding style by running `npm run format:check`
- [X] I have checked that my code contains no linting errors by running `npm run lint`
- [X] I have checked that my code contains no type errors by running `npm run typecheck`
- [X] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [X] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java [hunk 1]
```diff
@@ -38,8 +40,8 @@ protected JsonMergePatch readInternal(Class<? extends JsonMergePatch> clazz, Htt
 
 	@Override
 	protected void writeInternal(JsonMergePatch jsonMergePatch, HttpOutputMessage httpOutputMessage) throws IOException, HttpMessageNotWritableException {
-		try {
-			Json.createWriter(httpOutputMessage.getBody()).write(jsonMergePatch.toJsonValue());
+		try (JsonWriter jsonWriter = Json.createWriter(httpOutputMessage.getBody())){
```
Comment (gregory-j-baker) [inline backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java:43]:
```suggestion
		try (final var jsonWriter = Json.createWriter(httpOutputMessage.getBody())){
```

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java [hunk 2]
```diff
@@ -27,8 +29,8 @@ protected boolean supports(Class<?> clazz) {
 
 	@Override
 	protected JsonMergePatch readInternal(Class<? extends JsonMergePatch> clazz, HttpInputMessage httpInputMessage) throws IOException, HttpMessageNotReadableException {
-		try {
-			return Json.createMergePatch(Json.createReader(httpInputMessage.getBody()).readValue());
+		try (JsonReader jsonReader = Json.createReader(httpInputMessage.getBody())) {
```
Comment (gregory-j-baker) [inline backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java:32]:
```suggestion
		try (final var jsonReader = Json.createReader(httpInputMessage.getBody())) {
```

**Review by gregory-j-baker — CHANGES_REQUESTED** [review]
Trivial syntactic changes requested, otherwise approved.

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java (hunk 1 above)
Comment (CanadianPhilipBarkow) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java]: Implemented in commit 74e3d8d4dbedcbce2497740fdcf756f4ac2d3d92.

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java (hunk 2 above)
Comment (CanadianPhilipBarkow) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/json/JsonMergePatchHttpMessageConverter.java]: Implemented in commit 74e3d8d4dbedcbce2497740fdcf756f4ac2d3d92.

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**